File: ProtoType Pattern/PHP/SoftCopyBook.php
<?php

class SoftCopyBook extends Book
{

    /**
     * HardCopyBook constructor.
     */
    public function __construct($name,$description,$price)
    {
        parent::__construct($name,$description,$price);
    }

    public function __toString()
    {
        return parent::__toString();
    }

}

File: ProtoType Pattern/Java/src/dev/ismail/SoftCopyBook.java
package dev.ismail;

public class SoftCopyBook extends Book {


    public SoftCopyBook(String name, String description, int price) {
        super(name, description, price);
    }


    @Override
    protected Book clone() throws CloneNotSupportedException {
        return (SoftCopyBook) super.clone();
    }

    @Override
    public String toString() {
        return super.toString();
    }
}
